packages/core/src/shared/utilities/collectionUtils.ts

```diff
@@ -304,26 +305,36 @@ export function assign<T extends Record<any, any>, U extends Partial<T>>(data: T
  * @param depth
  * @param omitKeys Omit properties matching these names (at any depth).
  * @param replacement Replacement for object whose fields extend beyond `depth`, and properties matching `omitKeys`.
+ * @param maxStringLength truncates string values that exceed this threshold.
  */
-export function partialClone(obj: any, depth: number = 3, omitKeys: string[] = [], replacement?: any): any {
+export function partialClone(
+    obj: any,
+    depth: number = 3,
+    omitKeys: string[] = [],
+    options?: {
+        replacement?: any
+        maxStringLength?: number
+    }
+): any {
     // Base case: If input is not an object or has no children, return it.
     if (typeof obj !== 'object' || obj === null || 0 === Object.getOwnPropertyNames(obj).length) {
-        return obj
+        const maxLength = options?.maxStringLength
+        return typeof obj === 'string' && maxLength !== undefined ? truncate(obj, maxLength, '...') : obj
```

Comment:
nit: move this to an if condition since we may want to add manipulations for other types in the future